src/tax-lot/tax-lot.repository.ts

```diff
@@ -32,7 +37,7 @@ export class TaxLotRepository {
     }
   }
 
-  async findByBbl(bbl: string) {
+  async findByBbl(bbl: string): Promise<FindByBbl | undefined> {
```

Comment:
Using the inferred type from the Zod Schema on the repository method helps statically confirm the query matches the zod schema.

We could hypothetically also use the actual zod shema for run time validation. However, I do not think it is necessary. During development, manual testing and static type checking  should catch any run time errors.